spec/src/main/asciidoc/repository.asciidoc

```diff
@@ -460,53 +460,68 @@ create table Book (
+
+Furthermore, within the context of a given entity, each persistent field of an embeddable class reachable by navigation from the entity class may be assigned a compound name. The compound name is obtained by concatenating the names assigned to each field traversed by navigation from the entity class to the persistent field of the embedded class, optionally joined by a delimiter.
+
+The rule for concatenating compound names depends on the context, and is specified in <<property-name-concatenation>>. The examples in the table assume an `Order` entity has an `address` of type `MailingAddress` with a `zipCode` of type `int`.
+
+[[property-name-concatenation]]
+.Field Name Concatenation and Delimiters
+[cols="3, 2, ^1, 6"]
 |===
-|Entity Property Location |Delimiter |Example
+| Context | Type | Delimiter | Example
 
-|`@Query` annotation
+| `@Find` | Parameter name
+|`_`
+|`@Find List<Order> find(int address_zipCode);`
+
+|`@Query` | Path expression within query
 |`.`
 |`@Query("SELECT o FROM Order o WHERE o.address.zipCode=?1")`
 
-|_Query by Method Name_ method name
+|_Query by Method Name_ | Method name
 |`_`
 |`List<Order> findByAddress_zipCode(int zip);`
 
-|_Parameter-based Conditions_ parameter name
-|`_`
-|`@Find List<Order> find(int address_zipCode);`
-
-|`Sort` property value
+|`Sort` | String argument
 |`.` or `_`
 |`Sort.asc("address_zipCode")`
 
-|`@By` or `@OrderBy` annotation value
+|`@By` or `@OrderBy` | Annotation value
 |`.` or `_`
 |`@Find List<Order> find(@By("address.zipCode") int zip);`
 |===
 
-For a given entity property name, delimiter usage must be consistent. Either the delimiter must always be used within the entity property name to delimit subcomponents or the delimiter must never be used within the entity property name. Except in the case of `@Query` where the delimiter is required by the query language, delimiters can be omitted entirely from an entity property name when it is unnecessary to disambiguate the entity property to which the name refers.
+NOTE: Application programmers are strongly encouraged to follow Java's camel case naming standard for fields of entities, relations, and embeddable classes, avoiding underscores in field names. The resolution algorithm for persistent field identification relies on the use of underscore as a delimiter. Adhering to the camel case naming convention ensures consistency and eliminates ambiguity.
+
+
+==== Persistent Field Names in Query by Method Name
 
-The resolution of properties involves the following steps:
+For _Query by Method Name_, the use of delimiters within a compound name is optional. Delimiters may be omitted entirely from a compound name when they are not needed to disambiguate the persistent field to which the name refers. But for a given entity property name, delimiter usage must be consistent: either the delimiter must be used between every pair of persistent field names within the compound name, or it must not occur within the compound name.
```

Comment:
This can be taken out of this section because it belongs above (see prior comment) where it applies to the other patterns as well as Query by Method Name

```suggestion
For _Query by Method Name_, the use of delimiters within a compound name is optional. Delimiters may be omitted entirely from a compound name when they are not needed to disambiguate the persistent field to which the name refers.
```